mongoose/index.js

```diff
@@ -68,19 +68,24 @@ const mongooseExtendOptions = (Schema) => {
 
 		// Set option defaults
 		if (options.limit == null) options.limit = LIMIT;
-		if (options.populate == null) options.populate = POPULATE;
-		if (options.projection == null) options.projection = PROJECTION;
 		if (options.skip == null) options.skip = SKIP;
-		if (options.sortOrder == null) options.sortOrder = SORT_ORDER;
-		if (options.sortProperty == null) options.sortProperty = SORT_PROPERTY;
 
 		// Alter query from options
 		if (options.limit !== 0) {
 			query = query.skip(Number(options.skip)).limit(Number(options.limit));
 		}
-		query = query.populate(options.populate);
-		query = query.select(options.projection);
-		query = query.sort({ [options.sortProperty]: options.sortOrder });
+
+		if (options.populate) {
+			query = query.populate(options.populate);
+		}
+
+		if (options.projection) {
+			query = query.select(options.projection);
+		}
+
+		if (options.sortProperty && options.sortOrder) {
```

Comment:
Did you double check that these 2 are always used together?

Reply:
If you look at the syntax below you'll see they are both necessary.  Could default to 1 on sortOrder, though.  Anybody else have thoughts on this?

Reply:
Yes, but there could be existing code that doesn't provide both if they counted on the 1 being set by default for order.

Reply:
I think you want to maintain a default setting for sortOrder, but only apply the sort function if a sortProperty is passed.

Reply:
Okay, added in the default sort order